src/Akka.HealthCheck.Persistence/AkkaPersistenceLivenessProbe.cs

```diff
@@ -156,20 +168,35 @@ private bool HandleMessages(object message)
                     _probe = null;
                     if(_logInfo)
                         _log.Debug($"Persistence probe terminated. Recreating in {_delay.TotalSeconds} seconds.");
+                    
+                    Timers.CancelAll();
                     ScheduleProbeRestart();
                     return true;
                 
                 case CreateProbe:
                     if(_logInfo)
                         _log.Debug("Recreating persistence probe.");
                     
+                    Timers.StartSingleTimer(TimeoutTimerKey, CheckTimeout.Instance, _timeout);
                     _probe = Context.ActorOf(Props.Create(() => new SuicideProbe(Self, _probeCounter == 0, _id)));
                     Context.Watch(_probe);
                     _probe.Tell("hit" + _probeCounter);
                     _probeCounter++;
                     return true;
                 
+                case CheckTimeout:
+                    const string errMsg = "Timeout while checking persistence liveness. Recovery status is undefined.";
+                    _log.Warning(errMsg);
+                    _currentLivenessStatus = new PersistenceLivenessStatus(errMsg);
+                    PublishStatusUpdates();
+                    
+                    if(_probe is not null)
+                        Context.Stop(_probe);
```

Comment:
Force stop the suicide actor if it stalls/deadlocks

Reply:
Note that we also emit a new liveness status with a timeout message to let subscriber know about the problem.